s3prl/downstream/runner.py

```diff
@@ -448,64 +496,76 @@ def inference(self):
 
     def push_to_huggingface_hub(self):
+        """Creates a downstream repository on the Hub and pushes training artifacts to it."""
         if self.args.hf_hub_org.lower() != "none":
             organization = self.args.hf_hub_org
         else:
             organization = os.environ.get("HF_USERNAME")
         huggingface_token = HfFolder.get_token()
         print(f"[Runner] - Organisation to push fine-tuned model to: {organization}")
         
-        # Create repo on the Hub
-        upstream_model = self.args.upstream.replace("/", "__")
-        repo_name = f"superb-s3prl-{upstream_model}-{self.args.downstream}-{str(uuid.uuid4())[:8]}"
+        # Extract upstream repository metadata
+        if self.args.hub == "huggingface":
+            model_info = HfApi().model_info(self.args.upstream, token=huggingface_token)
+            downstream_model_id = model_info.sha
+            # Exclude "/" characters from downstream repo ID
+            upstream_model_id = model_info.modelId.replace("/", "__")
+        else:
+            upstream_model_id = self.args.upstream.replace("/", "__")
+            downstream_model_id = str(uuid.uuid4())[:8]
+        repo_name = f"{upstream_model_id}__{downstream_model_id}"
+        # Create downstream repo on the Hub
         repo_url = HfApi().create_repo(
             token=huggingface_token,
             name=repo_name,
             organization=organization,
             exist_ok=True,
-            private=True,
+            private=False,
         )
         print(f"[Runner] - Created Hub repo: {repo_url}")
 
         # Download repo and copy templates
         HF_HUB_DIR = "hf_hub"
-        REPO_PATH = os.path.join(self.args.expdir, HF_HUB_DIR, repo_name)
+        REPO_ROOT_DIR = os.path.join(self.args.expdir, HF_HUB_DIR, repo_name)
+        REPO_TASK_DIR = os.path.join(REPO_ROOT_DIR, self.args.downstream)
+        print(f"[Runner] - Cloning Hub repo to {REPO_ROOT_DIR}")
         model_repo = Repository(
-            local_dir=REPO_PATH, clone_from=repo_url, use_auth_token=huggingface_token
+            local_dir=REPO_ROOT_DIR, clone_from=repo_url, use_auth_token=huggingface_token
         )
+        # Pull latest changes if they exist
+        model_repo.git_pull()
         TEMPLATES_PATH = Path(f"./downstream/{self.args.downstream}/hf_hub_templates/")
         if TEMPLATES_PATH.exists():
-            shutil.copytree(TEMPLATES_PATH, REPO_PATH, dirs_exist_ok=True)
+            shutil.copytree(TEMPLATES_PATH, REPO_TASK_DIR, dirs_exist_ok=True)
         else:
             print(f"[Runner] - No Hugging Face Hub template found for downstream task! Experiment files will still be pushed to the Hub in raw form")
 
         # Copy checkpoints, tensorboard logs, and args / configs
-        shutil.copytree(self.args.expdir, REPO_PATH, dirs_exist_ok=True, ignore=shutil.ignore_patterns(HF_HUB_DIR))
-
-        # Inject upstream model name into model card
```

Comment:
I removed this step since I don't think it is need any longer

Reply:
OK!